error: unknown type name 'Evaluator'

Hi All,

根据guideline   http://www.paddlepaddle.org/docs/develop/documentation/zh/mobile/cross_compiling_for_android_cn.html
我尝试在mac上编译armeabi-v7a，Android API 21的PaddlePaddle库
使用命令行：
```
$ docker run -it --rm -v $PWD:/paddle -e "ANDROID_ABI=armeabi-v7a" -e "ANDROID_API=21" username/paddle-android:dev
```
但是出现如下错误
```
/paddle/paddle/trainer/Tester.h:82:24: error: unknown type name 'Evaluator'
                       Evaluator* evaluator,
                       ^
/paddle/paddle/trainer/Tester.h:91:19: error: use of undeclared identifier
      'ParameterClient2'
  std::shared_ptr<ParameterClient2> testParameterClient_;
                  ^
/paddle/paddle/trainer/Tester.h:96:19: error: use of undeclared identifier
      'Evaluator'
  std::unique_ptr<Evaluator> testEvaluator_;
                  ^
In file included from /paddle/paddle/gserver/tests/test_CompareSparse.cpp:17:
In file included from /paddle/paddle/trainer/Trainer.h:31:
/paddle/paddle/trainer/TrainerInternal.h:108:35: error: unknown type name
      'Evaluator'
  inline void setCurrentEvaluator(Evaluator* eval) { currentEvaluator_ = eval; }
                                  ^
/paddle/paddle/trainer/TrainerInternal.h:114:28: error: unknown type name
      'Evaluator'
  inline void setEvaluator(Evaluator* eval) { evaluator_ = eval; }
                           ^
/paddle/paddle/trainer/TrainerInternal.h:135:3: error: unknown type name
      'Evaluator'
  Evaluator* currentEvaluator_;
  ^
/paddle/paddle/trainer/TrainerInternal.h:136:3: error: unknown type name
      'Evaluator'
  Evaluator* evaluator_;
  ^
In file included from /paddle/paddle/gserver/tests/test_CompareSparse.cpp:17:
/paddle/paddle/trainer/Trainer.h:185:19: error: use of undeclared identifier
      'Evaluator'
  std::unique_ptr<Evaluator> evaluator_;
                  ^
/paddle/paddle/trainer/Trainer.h:186:19: error: use of undeclared identifier
      'Evaluator'
  std::unique_ptr<Evaluator> currentEvaluator_;
                  ^
/paddle/paddle/trainer/Trainer.h:187:19: error: use of undeclared identifier
      'Evaluator'
  std::unique_ptr<Evaluator> averageEvaluator_;
                  ^
In file included from /paddle/paddle/gserver/tests/test_CompareSparse.cpp:20:
/paddle/paddle/pserver/ParameterServer2.h:37:10: fatal error: 
      'ParameterService.pb.h' file not found
#include "ParameterService.pb.h"
         ^
11 errors generated.
paddle/gserver/tests/CMakeFiles/test_CompareSparse.dir/build.make:62: recipe for target 'paddle/gserver/tests/CMakeFiles/test_CompareSparse.dir/test_CompareSparse.cpp.o' failed
make[2]: *** [paddle/gserver/tests/CMakeFiles/test_CompareSparse.dir/test_CompareSparse.cpp.o] Error 1
CMakeFiles/Makefile2:2667: recipe for target 'paddle/gserver/tests/CMakeFiles/test_CompareSparse.dir/all' failed
make[1]: *** [paddle/gserver/tests/CMakeFiles/test_CompareSparse.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 83%] Building CXX object paddle/gserver/tests/CMakeFiles/test_RecurrentLayer.dir/test_RecurrentLayer.cpp.o
[ 83%] Linking CXX executable test_RecurrentLayer
[ 83%] Built target test_RecurrentLayer
```
我的MACIOS版本：10.13.1
我的Docker版本：17.09.0-ce

Thanks,

--- a/paddle/gserver/tests/CMakeLists.txt
+++ b/paddle/gserver/tests/CMakeLists.txt
@@ -62,11 +62,11 @@ if(NOT WITH_DOUBLE AND NOT MOBILE_INFERENCE)
 endif()
 
 if(NOT MOBILE_INFERENCE)
-################## test_Evaluator #######################
+    ################## test_Evaluator #######################
     add_unittest(test_Evaluator
         test_Evaluator.cpp)
       
-############### test_RecurrentGradientMachine ###############
+    ############### test_RecurrentGradientMachine ###############
     # TODO(yuyang18): There is some bug in test_RecurrentGradientMachine
     # I will fix it.
     add_unittest_without_exec(test_RecurrentGradientMachine
@@ -77,7 +77,7 @@ if(NOT MOBILE_INFERENCE)
                 ${CMAKE_CURRENT_BINARY_DIR}/test_RecurrentGradientMachine
         WORKING_DIRECTORY ${PADDLE_SOURCE_DIR}/paddle)
       
-############### test_NetworkCompare ###############
+    ############### test_NetworkCompare ###############
     add_unittest_without_exec(test_NetworkCompare
         test_NetworkCompare.cpp)
     if(WITH_GPU)
@@ -89,34 +89,33 @@ if(NOT MOBILE_INFERENCE)
             COMMAND .set_python_path.sh -d ${PADDLE_SOURCE_DIR}/python ${CMAKE_CURRENT_BINARY_DIR}/test_NetworkCompare --use_gpu=false
             WORKING_DIRECTORY ${PADDLE_SOURCE_DIR}/paddle)
     endif()
+
+    ################# test_CompareSparse ##################
+    add_unittest_without_exec(test_CompareSparse
+        test_CompareSparse.cpp)
+    if(NOT ON_TRAVIS)
+      add_test(NAME test_CompareSparse
+        COMMAND ${PADDLE_SOURCE_DIR}/paddle/.set_python_path.sh -d
+              ${PADDLE_SOURCE_DIR}/python:${PADDLE_SOURCE_DIR}/paddle/gserver/tests
+                  ./.set_port.sh -p port -n 6
+                      ${CMAKE_CURRENT_BINARY_DIR}/test_CompareSparse
+        WORKING_DIRECTORY ${PADDLE_SOURCE_DIR}/paddle/)
+    endif()
+
+    ################ test_CompareTwoNets ######################
+    add_unittest_without_exec(test_CompareTwoNets
+        test_CompareTwoNets.cpp)
+    add_test(NAME test_CompareTwoNets
+      COMMAND ${PADDLE_SOURCE_DIR}/paddle/.set_python_path.sh -d
+            ${PADDLE_SOURCE_DIR}/python:${PADDLE_SOURCE_DIR}/paddle/gserver/tests
+            ${CMAKE_CURRENT_BINARY_DIR}/test_CompareTwoNets
+        WORKING_DIRECTORY ${PADDLE_SOURCE_DIR}/paddle/)
 endif()
 
-
+################ test_PyDataProvider2 ######################
 add_unittest_without_exec(test_PyDataProvider2
         test_PyDataProvider2.cpp)
-
 add_test(NAME test_PyDataProvider2
    COMMAND .set_python_path.sh -d ${PADDLE_SOURCE_DIR}/paddle/gserver/tests:${PADDLE_SOURCE_DIR}/python ${CMAKE_CURRENT_BINARY_DIR}/test_PyDataProvider2
         WORKING_DIRECTORY ${PADDLE_SOURCE_DIR}/paddle
 )
-
-################# test_CompareSparse ##################
-add_unittest_without_exec(test_CompareSparse
-    test_CompareSparse.cpp)
-if(NOT ON_TRAVIS)
-  add_test(NAME test_CompareSparse
-    COMMAND ${PADDLE_SOURCE_DIR}/paddle/.set_python_path.sh -d
-          ${PADDLE_SOURCE_DIR}/python:${PADDLE_SOURCE_DIR}/paddle/gserver/tests
-              ./.set_port.sh -p port -n 6
-                  ${CMAKE_CURRENT_BINARY_DIR}/test_CompareSparse
-    WORKING_DIRECTORY ${PADDLE_SOURCE_DIR}/paddle/)
-endif()
-
-################ test_CompareTwoNets ######################
-add_unittest_without_exec(test_CompareTwoNets
-    test_CompareTwoNets.cpp)
-add_test(NAME test_CompareTwoNets
-  COMMAND ${PADDLE_SOURCE_DIR}/paddle/.set_python_path.sh -d
-        ${PADDLE_SOURCE_DIR}/python:${PADDLE_SOURCE_DIR}/paddle/gserver/tests
-        ${CMAKE_CURRENT_BINARY_DIR}/test_CompareTwoNets
-    WORKING_DIRECTORY ${PADDLE_SOURCE_DIR}/paddle/)

